### PR TITLE
[MMCA-4999] - Implement Start date & End date field errors on cash account transaction request journey

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Code changes to implement the date error scenarios with more specific ones

Below have been done as part this PR

- [x] add new tests and update the existing tests
- [x] relevant code changes
- [x] new message keys
- [x] refactoring in SelectLocalDateFormatter by reordering the methods and making reusable methods
- [x] other minor refactoring
- [x] fix test in ConfirmationPageSpec

### CODE REVIEW

File: test/forms/SelectTransactionsFormProviderSpec.scala [hunk 1]
```diff
@@ -251,11 +282,17 @@ class SelectTransactionsFormProviderSpec extends SpecBase {
     val startKey = "start"
     val endKey = "end"
 
-    val invalidMsgStartKey = "cf.form.error.start.date-number-invalid"
-    val invalidDateMsgStartKey = "cf.form.error.start.date.invalid.real-date"
+    val emptyMonthStartDateErrorKey = "cf.cash-account.transactions.request.start.date.empty.month"
```
Comment (Neozxc) [inline test/forms/SelectTransactionsFormProviderSpec.scala:285]: Should we wrap these messages with `messages()` for localization?

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala [hunk 2]
```diff
@@ -35,29 +35,42 @@ class SelectLocalDateFormatterSpec extends SpecBase {
...
+        Left(Seq(FormError("start.month", List(emptyMonthMsgKey), List())))
     }
 
     "return the correct FormError with keys when the supplied data empty year" in new SetUp {
       localDateFormatter.bind(key, bindDataDateWithEmptyYear) shouldBe
-        Left(Seq(FormError("start.year", List(invalidMsgKey), List())))
+        Left(Seq(FormError("start.year", List(emptyYearMsgKey), List())))
     }
 
     "return the correct FormError with keys when the supplied data is invalid month" in new SetUp {
```
Comment (Neozxc) [inline test/forms/mappings/SelectLocalDateFormatterSpec.scala:54]: Could we remove this space here please, to be in align with other tests, i know you haven't added this

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala [hunk 3]
```diff
@@ -143,40 +156,52 @@ class SelectLocalDateFormatterSpec extends SpecBase {
     val day: Int = 1
 
     val key = "start"
-    val invalidMsgKey = "cf.form.error.start.date-number-invalid"
-    val monthMsgKey = "cf.form.error.start.date.invalid.month"
-    val yearMsgKey = "cf.form.error.start.date.invalid.year"
-    val invalidRealDateMsgKey = "cf.form.error.start.date.invalid.real-date"
+    val emptyMonthAndYearMsgKey = "cf.cash-account.transactions.request.start.date.empty.month.year"
```
Comment (Neozxc) [inline test/forms/mappings/SelectLocalDateFormatterSpec.scala:159]: Should we wrap these messages with `messages()` for localization?

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala [hunk 4]
```diff
@@ -143,40 +156,52 @@ class SelectLocalDateFormatterSpec extends SpecBase {
...
+    val emptyMonthMsgKey = "cf.cash-account.transactions.request.start.date.empty.month"
+    val emptyYearMsgKey = "cf.cash-account.transactions.request.start.date.empty.year"
+    val invalidDateMsgKey = "cf.cash-account.transactions.request.start.date.invalid"
 
     val bindDataValid: Map[String, String] = Map(
       "start.day" -> day.toString, "start.month" -> "10", "start.year" -> "2022")
 
     val bindDataEmptyDate: Map[String, String] = Map(
```
Comment (Neozxc) [inline test/forms/mappings/SelectLocalDateFormatterSpec.scala:167]: Could we remove this please bindDataEmptyDate, this currently is not being used anywhere? 
Could you also please remove them as they are also not being used:

val year1
val bindDataInValidDate

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply test/forms/mappings/SelectLocalDateFormatterSpec.scala:54]: Done

File: test/forms/SelectTransactionsFormProviderSpec.scala (hunk 1 above)
Comment (tyagikumarvineet) [reply test/forms/SelectTransactionsFormProviderSpec.scala:285]: No @Neozxc, as we are dealing only with the keys and the localization will take place in the view or in the controller (if used)

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala (hunk 3 above)
Comment (tyagikumarvineet) [reply test/forms/mappings/SelectLocalDateFormatterSpec.scala:159]: No @Neozxc, as we are only dealing and checking the message keys' name and the localization will take place in the view or in the controller (if used)

**Review by Neozxc — APPROVED** [review]

**Pull request merged**